### PR TITLE
Rename Alternator batch item count metrics

### PR DESCRIPTION
**Pull request by amnonh**
This PR addresses multiple issues with alternator batch metrics:

1. Rename the metrics to scylla_alternator_item_count with op=BatchGetItem/BatchWriteItem
2. The batch size calculation was wrong and didn't count all items in the batch.
3. Add a test to validate that the metrics values increase by the correct value (not just increase). This also requires an addition to the testing to validate multiple metrics and an exact value change.

Needs backporting to allow the monitoring to use the correct metrics names.

### CODE REVIEW

File: alternator/executor.cc [hunk 1]
```diff
@@ -2226,6 +2226,7 @@ future<executor::request_return_type> executor::batch_write_item(client_state& c
                     co_return api_error::validation("Provided list of item keys contains duplicates");
                 }
                 used_keys.insert(std::move(mut_key));
+                batch_size++;
```
Comment (mykaul) [inline alternator/executor.cc:2229]: nit: can't we just use `batch_size += used_keys.size()` after we exit the inner loop adding the items?

Comment (amnonh) [reply alternator/executor.cc:2229]: It will not have the same all-or-nothing effect incase of validation error

File: alternator/executor.cc [hunk 2]
```diff
@@ -2216,6 +2215,7 @@ future<executor::request_return_type> executor::batch_write_item(client_state& c
...
```
Comment (nyh) [inline alternator/executor.cc:2218]: I didn't understand the commit message. You used the word "inaccurate" and "incorrect", but what was actually incorrect? I was worried in my review that this code actually counts the number of tables, not the number of items. Is this it? If so, please say so and don't say "inaccurate". If it's not it, please explain what was the inaccuracy. Thanks.

File: alternator/executor.cc (hunk 1 above)
Comment (nyh) [reply alternator/executor.cc:2229]: @amnon, I don't understand - when there's a validation error, NO item gets written or read at all (the whole operation is ignored), so we don't want to increase the counter at all. Do you do that? It will be nice to have a test for that as well.

File: alternator/executor.cc (hunk 2 above)
Comment (amnonh) [reply alternator/executor.cc:2218]: I'll change the commit message to make what was wrong clearer.

File: test/alternator/test_metrics.py [hunk 3]
```diff
@@ -119,15 +119,15 @@ def check_increases_operation(metrics, operation_names, metrics_names = ['scylla
 ###### Test for metrics that count DynamoDB API operations:
 
 def test_batch_write_item(test_table_s, metrics):
-    with check_increases_operation(metrics, ['BatchWriteItem']):
+    with check_increases_operation(metrics, ['BatchWriteItem'], metrics_names=['scylla_alternator_operation', 'scylla_alternator_item_count'], expected_values=lambda m, op: 1 if m == 'scylla_alternator_operation' else 2):
```
[the remark this answers is not in the document]
Comment (amnonh) [reply test/alternator/test_metrics.py:122]: Wouldn't you read scylla_alternator_operation_item_count as the number of operations? in this case we are talking about operations that are inside batches and maybe inside other things that acts like batches

File: alternator/stats.cc [hunk 4]
```diff
@@ -98,6 +96,10 @@ stats::stats() : api_operations{} {
                     seastar::metrics::description("number of rows read and matched during filtering operations")),
             seastar::metrics::make_total_operations("filtered_rows_dropped_total", [this] { return cql_stats.filtered_rows_read_total - cql_stats.filtered_rows_matched_total; },
                     seastar::metrics::description("number of rows read and dropped during filtering operations")),
+                    seastar::metrics::make_counter("item_count", seastar::metrics::description("The total number of items processed across all batches"),{op("BatchWriteItem")},
```
[the remark this answers is not in the document]
Comment (amnonh) [reply alternator/stats.cc:99]: You had another comment about the name, I don't have a strong opinion about it. I can remove the batch from the description, but then, how do you describe that it's part of a collection of items (batch or batch-like)?

If you can suggest a name that you feel comfortable with it would be great.

File: alternator/executor.cc (hunk 1 above)
Comment (amnonh) [reply alternator/executor.cc:2229]: That's what I wrote; with the current implementation, when there is a validation error, we do not increase the batch_size. This functionality will be lost if I follow @mykaul suggestion

Comment (mykaul) [reply alternator/executor.cc:2229]: I don't see why. The addition (using the .size() function) should be post-validation.

Comment (amnonh) [reply alternator/executor.cc:2229]: it will be inside the outer loop, so it can still fail mid-way

Comment (nyh) [reply alternator/executor.cc:2229]: I don't follow... No matter how you count the items, you only add the count to a local variable, and the global counter is only incremented by the count after all the validation and counting is over.

**Comment by amnonh** [comment]
You are right, yaniv suggest to change it

**Review by nyh — APPROVED** [review]
Looks good, thanks.

I would have added to the new test functions some more of the additional corner cases that were mentioned during the review - such 1. two different tables (it's easy, you can easily add another table fixture, e.g., "test_table_sn" or whatever) and write to it too 2. A query with an validation error in the second item doesn't increase the batch item counter at all.

**Comment by amnonh** [comment]
test fail looks like: #19645
